src/Adapter/ContextStateManager.php

```diff
@@ -178,6 +180,36 @@ public function setShop(Shop $shop): self
         return $this;
     }
 
+    /**
+     * @param ShopConstraint $shopConstraint
+     *
+     * @return $this
+     */
+    public function setShopContext(ShopConstraint $shopConstraint): self
+    {
+        $this->saveContextField('shop');
```

Comment:
This method doesn't handle ShopGroup correctly, we need to update it and store the potential shop group ID

Reply:
what do you mean?
Its handled in line 205 like this
```
        if ($shopConstraint->getShopGroupId()) {
            Shop::setContext(Shop::CONTEXT_GROUP, $shopConstraint->getShopGroupId()->getValue());

            return $this;
        }
```
Whats missing?
Do you mean we should also change context->shop to some shop of that group or so?

Reply:
Sorry, I meant `saveContextField` and `restoreShopContext` do not to handle the group ID so far. The `setShopContext` handles overriding the value correctly indeed But the saving and restoration of this data was forgotten